pkg/config/helpers.go

```diff
@@ -0,0 +1,426 @@
+package config
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	libp2p_crypto "github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/viper"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+	baccrypto "github.com/bacalhau-project/bacalhau/pkg/lib/crypto"
+	"github.com/bacalhau-project/bacalhau/pkg/lib/network"
+	"github.com/bacalhau-project/bacalhau/pkg/logger"
+)
+
+func Get[T any](c Context, key string) (T, error) {
+	raw := c.User().Get(key)
```

Comment:
These global getters and methods should be part of the `config.Context`

Reply:
They cannot be because of generics

Reply:
I see. It would be nice if we just pass BacalhauConfig and avoid having these getters in general.